crates/core/src/runloops/mod.rs

```diff
@@ -226,51 +233,51 @@ pub fn start_clock_runloop(mut slot_time: u64) -> (Receiver<ClockEvent>, Sender<
 }
 
 fn start_geyser_runloop(
+    plugin_config_paths: Vec<PathBuf>,
     plugin_manager_commands_rx: Receiver<PluginManagerCommand>,
     subgraph_commands_tx: Sender<SubgraphCommand>,
     simnet_events_tx: Sender<SimnetEvent>,
     geyser_events_rx: Receiver<GeyserEvent>,
 ) -> Result<JoinHandle<Result<(), String>>, String> {
     let handle = hiro_system_kit::thread_named("Geyser Plugins Handler").spawn(move || {
-        let mut plugin_manager = vec![];
+        let mut plugin_manager = GeyserPluginManager::new();
+        let mut surfpool_plugin = None;
+
+        for plugin_config_path in plugin_config_paths.into_iter() {
+            let plugin_manifest_location = FileLocation::from_path(plugin_config_path);
+            let contents = plugin_manifest_location.read_content_as_utf8()?;
+            let result: serde_json::Value = json5::from_str(&contents).unwrap();
+            let (plugin_name, plugin_dylib_path) = match (result.get("name").map(|p| p.as_str()), result.get("libpath").map(|p| p.as_str())) {
+                (Some(Some(name)), Some(Some(libpath))) => (name, libpath),
+                _ => {
+                    let error = format!("Unable to retrieve dylib: {}", plugin_manifest_location);
+                    let _ = simnet_events_tx.send(SimnetEvent::ErrorLog(Local::now(), error.clone()));
```

Comment:
Note to me for future cleanup: we can convert these `SimnetEvent::ErrorLog`s to `SimnetEvent::log_error(msg)`, to prevent  importing Local and just tidy things up a bit

Reply:
Addressed in #212